### ISSUE TITLE
Feat: start the ollama server from the app

### ISSUE THREAD

**Issue by ahmedoubadi** [pepperoni21/ollama-rs#82]
I would like to request a feature that allows the Ollama server to be started directly within the app, without requiring the user to manually start the server separately. This would streamline the user experience and ensure that the server is always running when the app is in use, reducing potential errors or confusion during setup.

Use Case:

Currently, the process of starting the Ollama server requires users to run it externally. By integrating the server startup into the app's lifecycle, users can benefit from:

A more seamless onboarding experience.
Reduced setup complexity, especially for non-technical users.
Better error handling when the server is not running or stops unexpectedly.

**Comment by pepperoni21** [comment]
So the library would support installing ollama and running the server directly from the rust code? Sounds like a good idea but I don't really know a good way of doing this.
Making an interface between the go code and rust? Installing the server and running it from the command line?

**Comment by InAnYan** [comment]
No, installing is too much, just starting is enough.

However,

Pros of auto start:
- No need to start externally.

Cons of auto start:
- What is the lifetime of an ollama server? It will be probably tighted up to the lifetime of a Rust application. It would seem to be wrong that ollama server is running after the Rust program is done.
- What if there will be two instances of ollama server? You can check in your library if ollama is running, but what if I run ollama serve after that? Two instances of ollama - 2x memory consumption. Is it even allowed to have 2 ollamas? They can interfere with their databases and file system.

Considering the conses, I think it's better not to add an auto start. `ollama-rs` as I see is just a wrapper around ollama server API

**Comment by pepperoni21** [comment]
I agree, ollama runs as a daemon in most systems, and that really isn't the scope of an API wrapper
